exporter/clickhouseexporter/exporter_logs_json.go

```diff
@@ -45,15 +48,48 @@ func (e *logsJSONExporter) start(ctx context.Context, _ component.Host) error {
 	}
 
 	if e.cfg.shouldCreateSchema() {
-		if err := internal.CreateDatabase(ctx, e.db, e.cfg.database(), e.cfg.clusterString()); err != nil {
-			return err
+		if createDBErr := internal.CreateDatabase(ctx, e.db, e.cfg.database(), e.cfg.clusterString()); createDBErr != nil {
+			return createDBErr
+		}
+
+		if createTableErr := createLogsJSONTable(ctx, e.cfg, e.db); createTableErr != nil {
+			return createTableErr
 		}
+	}
+
+	err = e.detectSchemaFeatures(ctx)
+	if err != nil {
+		return fmt.Errorf("schema detection: %w", err)
+	}
+
+	return nil
+}
+
+const (
+	logsJSONColumnResourceAttributesKeys = "ResourceAttributesKeys"
+	logsJSONColumnScopeAttributesKeys    = "ScopeAttributesKeys"
+	logsJSONColumnLogAttributesKeys      = "LogAttributesKeys"
+)
+
+func (e *logsJSONExporter) detectSchemaFeatures(ctx context.Context) error {
+	columnNames, err := internal.GetTableColumns(ctx, e.db, e.cfg.database(), e.cfg.LogsTableName)
+	if err != nil {
+		return err
+	}
 
-		if err := createLogsJSONTable(ctx, e.cfg, e.db); err != nil {
-			return err
+	for _, name := range columnNames {
+		switch name {
+		case logsJSONColumnResourceAttributesKeys:
+			e.schemaFeatures.AttributeKeys = true
+		case logsJSONColumnScopeAttributesKeys:
+			e.schemaFeatures.AttributeKeys = true
+		case logsJSONColumnLogAttributesKeys:
+			e.schemaFeatures.AttributeKeys = true
 		}
 	}
 
+	e.renderInsertLogsJSONSQL()
```

Comment:
It looks like this function is handling both schema feature detection and SQL rendering. Would it be better to move the SQL rendering logic outside of the detectSchemaFeature function?